microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/AbstractAnnotationTypeReferencePropertiesProvider.java

```diff
@@ -51,29 +54,51 @@ protected SearchPattern createSearchPattern(String annotationName) {
 
 	@Override
 	public void collectProperties(SearchMatch match, SearchContext context, IProgressMonitor monitor) {
-		Object element = match.getElement();
-		if (element instanceof IAnnotatable && element instanceof IJavaElement) {
-			IJavaElement javaElement = (IJavaElement) element;
-			processAnnotation(javaElement, context, monitor);
-		}
-	}
-
-	protected void processAnnotation(IJavaElement javaElement, SearchContext context, IProgressMonitor monitor) {
+		IJavaElement javaElement = null;
 		try {
-			String[] names = getAnnotationNames();
-			IAnnotation[] annotations = ((IAnnotatable) javaElement).getAnnotations();
-			for (IAnnotation annotation : annotations) {
-				for (String annotationName : names) {
-					if (isMatchAnnotation(annotation, annotationName)) {
-						processAnnotation(javaElement, annotation, annotationName, context, monitor);
-						break;
-					}
+			Object element = match.getElement();
+			if (match instanceof TypeReferenceMatch) {

```

Comment:
I noticed that in the case where the match is from a method parameter or a Java field, they both cause the `match` instance to be a  `TypeReferenceMatch`. 

I noticed the following: 
If it is from a method parameter, localElement exists
If it is from a Java field, localElement does not exist

Reply:
create a private static class method Object getMatchedElement(SearchMatch match) which does that and add comments that you explained below.

Reply:
fixed